extensions/ONNX/Onnx/OnnxTextGenerator.cs

```diff
@@ -0,0 +1,178 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System.Diagnostics.CodeAnalysis;
+using System.Runtime.CompilerServices;
+using Microsoft.Extensions.Logging;
+using Microsoft.KernelMemory.AI.OpenAI;
+using Microsoft.KernelMemory.Diagnostics;
+using Microsoft.ML.OnnxRuntimeGenAI;
+using static Microsoft.KernelMemory.OnnxConfig;
+
+namespace Microsoft.KernelMemory.AI.Onnx;
+
+/// <summary>
+/// Text generator based on Onnx models, via OnnxRuntimeGenAi
+/// See https://github.com/microsoft/onnxruntime-genai
+/// </summary>
+[Experimental("KMEXP01")]
+public sealed class OnnxTextGenerator : ITextGenerator, IDisposable
+{
+    private readonly Model? _model = default;
+    private readonly Tokenizer? _tokenizer = default;
+    private readonly ILogger<OnnxTextGenerator> _log;
+    private readonly ITextTokenizer _textTokenizer;
```

Comment:
could you add some docs on _tokenizer and _textTokenizer? even after reading the code, it's not clear why there are two